drift_dev/lib/src/writer/manager_writer.dart

```diff
@@ -0,0 +1,625 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'package:drift_dev/src/analysis/results/results.dart';
+import 'package:drift_dev/src/writer/modules.dart';
+import 'package:drift_dev/src/writer/writer.dart';
+
+abstract class _FilterWriter {
+  /// The getter for the column on this table
+  ///
+  /// E.G `id` in `table.id`
+  final String fieldGetter;
+
+  /// The getter for the columns filter
+  ///
+  /// E.G `id` in `f.id.equals(5)`
+  final String filterName;
+
+  /// An abstract class for all filters
+  _FilterWriter(this.filterName, {required this.fieldGetter});
+
+  /// Write the filter to a provider [TextEmitter]
+  void writeFilter(TextEmitter leaf);
+}
+
+class _RegularFilterWriter extends _FilterWriter {
+  /// The type that this column is
+  ///
+  /// E.G `int`, `String`, etc
+  final String type;
```

Comment:
Typically we use `SqlType` (or, even better, `ColumnType` to support custom types) in the generator as well, you can use `leaf.innerColumnType(type)` if you need to write that type out. A downside of strings is that we loose metadata for imports (e.g. blob columns use `Uint8List` which require a `dart:typed_data` import in modular builds).

Reply:
Ah nevermind, you're already doing that when you're constructing the instance.